pkg/engine/masterarmresources.go

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+package engine
+
+import (
+	"github.com/Azure/aks-engine/pkg/api"
+	"github.com/Azure/go-autorest/autorest/to"
+)
+
+func createKubernetesMasterResources(cs *api.ContainerService) []interface{} {
+	var masterResources []interface{}
+
+	p := cs.Properties
+	//TODO: Implement CosmosEtcd
```

Comment:
Is this a non-trivial amount of work?

Reply:
As of now, yes. I will need to go with the monkey patching approach to get cosmosEtcd to work.